### ISSUE TITLE
add proxy-mode to default-modes crash

### ISSUE THREAD

**Issue by MasterCsquare** [atlas-engineer/next#325]
guix system, my init.lisp
```lisp
(in-package :next)

(setf (get-default 'proxy-mode 'server-address) "socks5://localhost:1080")
(add-to-default-list 'proxy-mode 'buffer 'default-modes)
```
the error messages
```
 <INFO> [23:32:21] next base.lisp (start) - NEXT::+VERSION+: "0.0.0" 
 <INFO> [23:32:21] next base.lisp (load-lisp-file form-fun-4) -
  Loading configuration from /home/mas/.config/next/init.lisp...
 <WARN> [23:32:21] next minibuffer.lisp (echo) -
  Can't echo 'MINIBUFFER-MODE enabled.' without minibuffer or interface
 <INFO> [23:32:21] next remote.lisp (initialize-instance :after remote-interface) -
  Bus connection name: :1.6879
 <INFO> [23:32:21] next port.lisp (run-program port) -
  Current directory: /home/mas/
 <INFO> [23:32:21] next port.lisp (run-program port) -
  Platform port path: /gnu/store/v68b0gla6lr51j9bhkfvmbv2w24xagq4-next-gtk-webkit-0.0.0/bin/next-gtk-webkit
 <INFO> [23:32:21] next port.lisp (run-program port) -
  Platform port arguments: NIL
 <INFO> [23:32:21] next port.lisp (run-program port) -
  Platform port log file: /home/mas/.local/share/next/gnu/store/v68b0gla6lr51j9bhkfvmbv2w24xagq4-next-gtk-webkit-0.0.0/bin/next-gtk-webkit.log
Unhandled DBUS/CONDITIONS:METHOD-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                    {10006205B3}>:
  Method error: "Message recipient disconnected from message bus without replying".

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10006205B3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<DBUS/CONDITIONS:METHOD-ERROR {1006B0DA93}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<DBUS/CONDITIONS:METHOD-ERROR {1006B0DA93}>)
2: (INVOKE-DEBUGGER #<DBUS/CONDITIONS:METHOD-ERROR {1006B0DA93}>)
3: (UIOP/IMAGE:HANDLE-FATAL-CONDITION #<DBUS/CONDITIONS:METHOD-ERROR {1006B0DA93}>)
4: (SB-KERNEL::%SIGNAL #<DBUS/CONDITIONS:METHOD-ERROR {1006B0DA93}>)
5: (ERROR DBUS/CONDITIONS:METHOD-ERROR :ARGUMENTS ("Message recipient disconnected from message bus without replying"))
6: (DBUS/MESSAGES:INVOKE-METHOD #<DBUS/TRANSPORT-UNIX:UNIX-CONNECTION {1006997B53}> "set_proxy" :PATH "/engineer/atlas/next/platform" :SIGNATURE "asssas" :ARGUMENTS (("1") "custom" "socks5://localhost:1080" ("localhost" "localhost:8080")) :INTERFACE "engineer.atlas.next.platform" :DESTINATION "engineer.atlas.next.platform" :NO-REPLY NIL :NO-AUTO-START NIL :ASYNCHRONOUS NIL :ENDIANNESS :LITTLE-ENDIAN)
7: (DBUS/CONVENIENCE::CALL-WITH-OPEN-BUS #<CLOSURE (LAMBDA (NEXT::BUS) :IN NEXT::%RPC-SEND) {1006996E1B}> #<event base, 1 FDs monitored, using: #<epoll(4) multiplexer> {10069954A3}> (#<DBUS/TRANSPORT-UNIX:UNIX-SERVER-ADDRESS {1006996CD3}>))
8: ((:METHOD NEXT::%RPC-SEND (NEXT::REMOTE-INTERFACE STRING)) #<unused argument> "set_proxy" ("1") "custom" "socks5://localhost:1080" ("localhost" "localhost:8080")) [fast-method]
9: ((:METHOD INITIALIZE-INSTANCE :AFTER (NEXT::PROXY-MODE)) #<NEXT::PROXY-MODE {1006990753}>) [fast-method]
10: ((SB-PCL::EMF INITIALIZE-INSTANCE) #<unused argument> #<unused argument> #<NEXT::PROXY-MODE {1006990753}> :NAME "PROXY-MODE" :BUFFER #<NEXT::BUFFER {1005B83753}> :BUFFER #<NEXT::BUFFER {1005B83753}> :ACTIVATE T)
11: ((:METHOD MAKE-INSTANCE (CLASS)) #<STANDARD-CLASS NEXT::PROXY-MODE> :NAME "PROXY-MODE" :BUFFER #<NEXT::BUFFER {1005B83753}> :BUFFER #<NEXT::BUFFER {1005B83753}> :ACTIVATE T) [fast-method]
12: ((:METHOD NEXT::PROXY-MODE (NEXT::ROOT-MODE)) #<unused argument> :BUFFER #<NEXT::BUFFER {1005B83753}> :ACTIVATE T) [fast-method]
13: ((:METHOD INITIALIZE-INSTANCE :AFTER (NEXT::BUFFER)) #<NEXT::BUFFER {1005B83753}>) [fast-method]
14: ((SB-PCL::EMF INITIALIZE-INSTANCE) #<unused argument> #<unused argument> #<NEXT::BUFFER {1005B83753}> :ID "1" :NAME "*Help*" :DEFAULT-MODES (NEXT::HELP-MODE NEXT::PROXY-MODE NEXT::DOCUMENT-MODE NEXT::ROOT-MODE))
15: ((:METHOD MAKE-INSTANCE (CLASS)) #<STANDARD-CLASS NEXT::BUFFER> :ID "1" :NAME "*Help*" :DEFAULT-MODES (NEXT::HELP-MODE NEXT::PROXY-MODE NEXT::DOCUMENT-MODE NEXT::ROOT-MODE)) [fast-method]
16: ((:METHOD NEXT::RPC-BUFFER-MAKE (NEXT::REMOTE-INTERFACE)) #<NEXT::REMOTE-INTERFACE {1004EFA263}> :NAME "*Help*" :DEFAULT-MODES (NEXT::HELP-MODE NEXT::PROXY-MODE NEXT::DOCUMENT-MODE NEXT::ROOT-MODE)) [fast-method]
17: ((:METHOD NEXT::HELP (NEXT::ROOT-MODE)) #<unused argument>) [fast-method]
18: ((:METHOD NEXT::INITIALIZE-PORT (NEXT::REMOTE-INTERFACE)) #<NEXT::REMOTE-INTERFACE {1004EFA263}> NIL) [fast-method]
19: (NEXT:ENTRY-POINT)
20: (NEXT-EXEC:MAIN)
21: ((LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE))
22: (UIOP/IMAGE:CALL-WITH-FATAL-CONDITION-HANDLER #<CLOSURE (LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE) {1004A6910B}>)
23: ((FLET SB-UNIX::BODY :IN SAVE-LISP-AND-DIE))
24: ((FLET "WITHOUT-INTERRUPTS-BODY-7" :IN SAVE-LISP-AND-DIE))
25: ((LABELS SB-IMPL::RESTART-LISP :IN SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
```
but **M-x proxy-mode** works.

**Comment by Ambrevar** [comment]
Thanks for reporting, I can reproduce.

I believe this is due to the platform port `set_proxy` endpoint which
segfaults when called on startup.

**Comment by Ambrevar** [comment]
Fixed in eefc5bf7f3425e82542a9c1fce93d6fe3176b7d4.

**Comment by MasterCsquare** [comment]
it works, great.
